numpy/core/meson.build

```diff
@@ -322,6 +321,17 @@ optional_function_attributes = [
 #  endif
 #endforeach
 
+# Max possible optimiztion flags
+# We pass this flags to all our dispatch-able
+# (multi_targets) sources.
+compiler = meson.get_compiler('c')
+compiler_id = compiler.get_id()
+max_opt = {
+  'msvc': ['/O2'],
+  'intel-cl': ['/O3'],
+}.get(compiler_id, ['-O3'])
```

Comment:
MSVC doesn't support `/O3` just `/O2`, get_option('optimization') affects on all the sources, we just want to target the multi-targets sources only, I can just add function attribute e.g. `optimization: str` to `features.multi_targets()` to access to underlying meson optimization flags, would it be okay?

Reply:
I'm not quite sure, I may be missing something - why do we need to force this? It's normally already set to the maximum optimization level on release builds (or more accurately: all builds that go through `meson-python`, also editable installs). And to `O2` for `buildtype=debugoptimized`. Users can deviate by explicitly choosing another optimization level. Why would we override here what is explicitly chosen?

Reply:
> I'm not quite sure, I may be missing something - why do we need to force this?

Mainly to guarantee of having auto-vectorization across all functions since its usually activated under `-O3`.

> It's normally already set to the maximum optimization level on release builds

Not sure about the current state of meson build, but within distutils/distro the default flag is `-O2`.

> Why would we override here what is explicitly chosen?

Users may not aware that some functions that don't have SIMD implementation counting on it,
for example: the macros autovec macros within [loops_autovec.dispatch.c.src](https://github.com/numpy/numpy/blob/main/numpy/core/src/umath/loops_autovec.dispatch.c.src) counting on it.

Also to guarantee of having the same FP behavior,  compilers may procedure different machine code affects
on the precession.

So we have three options here:

- Replace it with e.g. `#pragma GCC optimize("O3")` on the top of each multi-targets source
  not sure about other compilers.
- Leave it as-is, and enhance it later through `features.multi_targets()` method.
- Erase it and count on the user-defined optimization, the issue here that `O3`
  will be applied a cross all sources so maybe this kind of high level of optimization
  bring issues to sources aren't covered by us or leads to increases the binary size without
  any performance gain.

Reply:
Okay, thanks for the details Sayed. I suggest that we leave it as is for now then. It should work, and in case there's an issue we'll find it in the 1.26.0 beta/rc releases soon enough.